core/api/workspace.go

```diff
@@ -339,6 +352,17 @@ type ProcessDescription struct {
 	Running  bool   `json:"running"`
 }
 
+type ComponentStatus struct {
+	ComponentID         string            `json:"componentId"`
+	Running             bool              `json:"running"`
+	EnvVars             map[string]string `json:"envVars"`
+	CPUPercent          float64           `json:"CPUPercent"`
+	CreateTime          int64             `json:"createTime"`
+	ResidentMemory      uint64            `json:"residentMemory"`
+	Ports               []uint32          `json:"ports"`
+	ChildrenExecutables []string          `json:"childrenExecutables"`
```

Comment:
Future looking: This is a case where we might care about the distinction between "components" and "resources". A process component may have child processes, which are not quite controlled components, but are resources worth being aware of. Needs more thought...

In the meantime, it's a little weird that we're only capturing the executable name of each child. If any, I'd have expected child pids. In that case, you could imagine creating a process "resource" from a pid.

Reply:
Yeah, I don't think that it's the right thing to capture in the long-term but I do think that it provides the most useful information right now.